### DISCUSSION TITLE
Example risk analysis import not working, no error shown, Ubuntu 22.04

### DISCUSSION THREAD

**Discussion by roughentomologyx** [monarc-project/MonarcAppFO#476] [Q&A]
Hey, 
i can not use anything lower then ubuntu 22.04 for reasons beyond my control.
Since i'm asking for help for a not officially supported OS version i'm asking here instead of opening an issue, i hope this is correct.
I still wanted to install the front office from source for trying it out.
Everything works well with the installation.

But when i try to import the risk analysis from https://www.monarc.lu/assets/files/monarc-training/en/MyPrintEN.json it does not work.
It says "The instance has been imported successfully" but nothing is happening. 
Can you hint me in a direction that helps me fix this? What files would i have to provide to help you helping me?
I tried with Monarc version v2.12.4 and v2.12.5-p3 and get the same problem without error.
Thanks in advance.

**Comment by ruslanbaidan** [comment]
Hello,
I think the issues is that 22.04 has installed by default php 8.1, which is currently not supported by Monarc app.
We are working on a new release to support it, but for now you can downgrade your php to 7.4.
You could follow the following manual and replace the 8.0 to 7.4:
https://blog.devops.dev/downgrade-php8-1-to-php8-0-or-php7-4-on-ubuntu-22-04-2fab4a6a3be3

**Reply by roughentomologyx** [reply]
Hey, thank you a lot for the answer. 
I downgraded php to 7.4 and still get the same behaviour.

**Reply by ruslanbaidan** [reply]
Could you tell what is the output of the command:
`php -c /etc/php/apache2/php.ini -v` ?
The other thing you could check the apache error logs, located in `/var/logs/apache2/error.log` by default.
The other questions, how long the import of MyPrint takes for you ?
Normally, when the message is shown "The instance has been imported successfully", the import was either successful neither there are missing project configuration steps. Does the rest of the functionality work for you? Creation of new objects, moving them to the analysis (instantiation), Knowledge Base management etc.

**Reply by roughentomologyx** [reply]
php -c /etc/php/apache2/php.ini -v

PHP 7.4.33 (cli) (built: Jan 13 2023 10:42:56) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.33, Copyright (c), by Zend Technologies

**Reply by roughentomologyx** [reply]
Apache does not return an error in his logs. For him it seems everything is fine. 
The import takes around 0.5 seconds.
I will try to verify if the rest of the functionality is working.

**Reply by roughentomologyx** [reply]
This is happening when i try to import assets. 
![image](https://user-images.githubusercontent.com/45817753/215786970-97976030-01f0-4427-ac6d-dbd14e9f0cfc.png)
Excuse me for posting this as a picture ( can't copy from webapp ). 

Offending line 353 from AssetImportService.php:  `?: $this->referentialTable->findByAnrAndUuid($anr, $referentialUuid);`

Apache error.log:  [php7:notice] [pid 9948] [client ] PHP Notice:  Trying to access array offset on value of type null in /var/lib/monarc/releases/MonarcAppFO-v2.12.5-p3/vendor/monarc/frontoffice/src/Service/AssetImportService.php on line 350, referer:

**Comment by ruslanbaidan** [comment]
The Import of MyPrint takes around 30 seconds, depending on the system configuration.
So 0.5 sec is the sign that it's not even started.
Regarding the asset import. What was the scenario ? To test it (as an option), first we need to export an asset, remove it from the library, and import (add again).
I can confirm that both of the functionalities work on the latest Monarc version.

**Reply by roughentomologyx** [reply]
This error happens when i do add an asset -> asset import center -> monarc library. 

Creating an asset myself, exporting it, deleting it and importing it via add an asset -> asset import center ->  monarc export file works fine.

**Reply by ruslanbaidan** [reply]
Thank you for the details. We are going to investigate the library import error.

Regarding the MyPrint Import. There is a last point on the [Troubleshooting page](https://www.monarc.lu/documentation/common-issues/). You could check your settings and increase the limits. Normally the default limit of files upload is 2Mb, but in this case there should be an error displayed. However this is something that needs to be checked.

**Reply by ruslanbaidan** [reply]
The issue of the assets import `asset -> asset import center -> monarc library.` is fixed.
You could run `git pull` and `./scripts/update-all.sh` from the project root to update.

**Reply by roughentomologyx** [reply]
Hey. I did a fresh install on a Ubuntu 20.04. I encountered the same problem again in this environment. 
(When i try to import the risk analysis from https://www.monarc.lu/assets/files/monarc-training/en/MyPrintEN.json nothing is happening but i get "The instance has been imported successfully", no error logs for apache) 

I ran Monarc in a VM and imported the example risk analysis there and it worked fine.
I did `chown -R www-data:www-data /var/lib/monarc` to mitigate any permission issues. 
Any ideas what i'm doing wrong?

**Reply by ruslanbaidan** [reply]
Were the necessary recommended php configuration settings applied, linked [here](https://github.com/monarc-project/MonarcAppFO/discussions/476#discussioncomment-4838822)?
[This is the exact link](https://github.com/monarc-project/MonarcAppFO/blob/master/vagrant/bootstrap.sh#L22-L26).

**Reply by roughentomologyx** [reply]
I did not before because i was confused where to put them. Now i did and it worked! 
(If this is happening to anyone else, it's in /etc/php/7.4/apache2/php.ini where you have to change this).
Thank you so much for your help!
